File: talker.py
#!/usr/bin/env python


import rospy
import sys
from std_msgs.msg import String

def talker():
	
    pub = rospy.Publisher('chatter', String, queue_size=1000)
    rate = int(sys.argv[1]) 
    rospy.init_node('talker', anonymous=True)
    rate = rospy.Rate(rate) 
    prev_time = rospy.get_rostime()
    count = 0
    while not rospy.is_shutdown():
        hello_str = "hello world %s" % rospy.get_time()
        now = rospy.get_rostime()
        diff_time = (now - prev_time).to_sec()
        count+=1
        if( diff_time >= 1.0):
			rospy.loginfo("{0} / {1} = {2}".format(count, diff_time, count/diff_time) )
			count=0
			prev_time = now
			
        pub.publish(hello_str)
        rate.sleep()

if __name__ == '__main__':
    try:
		if ( len(sys.argv) != 2 ):
			print 'First command line argument must be the publishing rate'
		else:	
			talker()
    except rospy.ROSInterruptException:
        pass

File: src/talker.cpp

#include "ros/ros.h"
#include "rate_test/SuperAwesome.h"
#include <sstream>


int main(int argc, char **argv)
{
	if( argc != 2)
	{
		std::cout<< "Expecting the publishing rate as first command line argument" << std::endl;
		return 1;
	}	
	const int PERIOD = atoi(argv[1]);

  ros::init(argc, argv, "talker");
  ros::NodeHandle n;
  

  ros::Publisher chatter_pub = n.advertise<rate_test::SuperAwesome>("chatter", 1000);
  ros::Rate loop_rate(PERIOD);

  int count = 0;
  int total_msg_count = 0;
  ros::Time prev_time = ros::Time::now();
  
   
  while (ros::ok() && total_msg_count < PERIOD*10)
  {
    rate_test::SuperAwesome msg;
    std::stringstream ss;
    ss << "hello world " << count;
    msg.data = ss.str();

	ros::Time curr_time = ros::Time::now();
	double diff_time = (curr_time - prev_time).toSec();
	count++;
	total_msg_count++;
	
	// show the rate once per second
  	if( diff_time >= 1.0)
  	{
		ROS_INFO("%d: %f / %f = %f", total_msg_count, (double)count, diff_time , count/diff_time);
		count = 0;
		prev_time = curr_time;
	}

    chatter_pub.publish(msg);
    ros::spinOnce();
    loop_rate.sleep();
  } 

  ROS_INFO("total_messages: %d", total_msg_count);

  return 0;
}

